### PR TITLE
leverage toBlocking for asyn API test, #241.

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
Signed-off-by: Pan Li <panli@microsoft.com>

### CODE REVIEW

File: src/test/java/com/microsoft/azure/spring/data/cosmosdb/repository/integration/QuestionRepositoryAsyncIT.java [hunk 1]
```diff
@@ -49,58 +49,50 @@ public void testSaveAsync() {
...
-                    Assert.assertEquals(((DocumentClientException) e).getStatusCode(), HttpStatus.SC_NOT_FOUND);
-                }
-        );
+        final Question question = this.repository.findByIdAsync(QUESTION.getId()).toBlocking().single();
+        Assert.assertEquals(question, QUESTION);
+
+        try {
+            this.repository.findByIdAsync(NOT_EXIST_ID).toCompletable().await();
+        } catch (RuntimeException e) {
+            final Throwable cause = e.getCause();
```
Comment (xscript) [inline src/test/java/com/microsoft/azure/spring/data/cosmosdb/repository/integration/QuestionRepositoryAsyncIT.java]: If it doesn't throw, this case will pass?

Comment (Incarnation-p-lee) [reply src/test/java/com/microsoft/azure/spring/data/cosmosdb/repository/integration/QuestionRepositoryAsyncIT.java]: good catch, defect here, should add assert_not_reached after findByIdAsync

File: src/test/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplateIT.java [hunk 2]
```diff
@@ -262,21 +265,21 @@ public void testPaginationAsync() {
     @Test
     public void testInsertAsync() {
         this.dbTemplate.deleteAll(personInfo.getCollectionName(), personInfo.getPartitionKeyNames());
-        this.dbTemplate.insertAsync(this.personInfo.getCollectionName(), TEST_PERSON_0, null)
-                .subscribe(p -> assertThat(p).isEqualTo(TEST_PERSON_0));
-        this.dbTemplate.insertAsync(this.personInfo.getCollectionName(), TEST_PERSON_1, null)
-                .subscribe(p -> assertThat(p).isEqualTo(TEST_PERSON_1));
+        Person inserted = this.dbTemplate.insertAsync(this.personInfo.getCollectionName(), TEST_PERSON_0, null)
```
Comment (sophiaso) [inline src/test/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplateIT.java]: Save  `this.personInfo.getCollectionName()` to a variable for reuse?

Comment (Incarnation-p-lee) [reply src/test/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplateIT.java]: make sense, will cleanup this in another PR.

**Pull request merged**